packages/react-query/src/__tests__/transition.test.tsx

```diff
@@ -0,0 +1,92 @@
+import { afterAll, beforeAll, describe, expect, it } from 'vitest'
+import * as React from 'react'
+import {
+  createRenderStream,
+  useTrackRenders,
+} from '@testing-library/react-render-stream'
+import { QueryClientProvider, useQuery } from '..'
+import { QueryCache } from '../index'
+import { createQueryClient, queryKey, sleep } from './utils'
+
+describe('react transitions', () => {
```

Comment:
cc @TkDodo I pulled in your test from #8444 and it seemingly works?

I had to fix it in 38b88645e85bde8e2ccc42eb3d57997e9eb8deaf but I think that's fine or?